code/datums/diseases/black_goo.dm

```diff
@@ -1,100 +1,106 @@
 		if(1)
 			if(H.stat == DEAD && stage_counter != stage)
-				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage One! Zombie transformation begins at Stage Four."))
+				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage One! Zombie transformation begins at Stage Three."))
 				stage_counter = stage
-			survive_mob_death = TRUE //changed because infection rate was REALLY horrible.
 			if(goo_message_cooldown < world.time )
-				if(prob(3))
-					to_chat(affected_mob, SPAN_DANGER("You feel really warm..."))
+				if(prob(5))
+					to_chat(affected_mob, SPAN_DANGER("You feel warm..."))
 					goo_message_cooldown = world.time + 100
+					stage_level = stage_level + 4
+				else if(prob(3))
+					to_chat(affected_mob, SPAN_DANGER("You can't trust them..."))
+					goo_message_cooldown = world.time + 100
+					stage_level = stage_level + 2
+
+		//stage 2 : Ague "Febrile"
 		if(2)
 			if(H.stat == DEAD && stage_counter != stage)
-				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage Two! Zombie transformation begins at Stage Four."))
-				stage_counter = stage
-			if(goo_message_cooldown < world.time)
-				if (prob(3)) to_chat(affected_mob, SPAN_DANGER("Your throat is really dry..."))
-				else if (prob(6)) to_chat(affected_mob, SPAN_DANGER("You feel really warm..."))
-				else if (prob(2)) H.vomit_on_floor()
-				goo_message_cooldown = world.time + 100
-		if(3)
-			if(H.stat == DEAD && stage_counter != stage)
-				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage Three! Zombie transformation begins at Stage Four, which will be soon."))
+				to_chat(H, SPAN_CENTERBOLD("Your zombie infection is now at Stage Two! Zombie transformation begins at Stage Three."))
 				stage_counter = stage
-			hidden = list(0,0)
-			//survive_mob_death = TRUE //even if host dies now, the transformation will occur.
-			H.next_move_slowdown = max(H.next_move_slowdown, 1)
 			if(goo_message_cooldown < world.time)
-				if (prob(3))
-					to_chat(affected_mob, SPAN_DANGER("You cough up some black fluid..."))
-					goo_message_cooldown = world.time + 100
-				else if (prob(6))
+				if (prob(5))
 					to_chat(affected_mob, SPAN_DANGER("Your throat is really dry..."))
-					goo_message_cooldown = world.time + 100
-				else if (prob(9))
+					stage_level = stage_level + 10
+				else if (prob(6))
 					to_chat(affected_mob, SPAN_DANGER("You feel really warm..."))
-					goo_message_cooldown = world.time + 100
-				else if(prob(5))
-					goo_message_cooldown = world.time + 100
+					stage_level = stage_level +5
+				else if (prob(2))
+					to_chat(affected_mob, SPAN_DANGER("You cough up some black fluid..."))
+					stage_level = stage_level + 20
+				else if (prob(2))
 					H.vomit_on_floor()
-		if(4)
+					stage_level = stage_level + 15
+				else if(prob(3))
+					to_chat(affected_mob, SPAN_DANGER("You can't trust them..."))
+					stage_level = stage_level + 4
+				goo_message_cooldown = world.time + 100
```

Comment:
use cooldown macros

Reply:
done was a pain to understand the logic but i manage to understood it thanks to fira.